src/lib/opal_parser.c

```diff
@@ -161,25 +181,28 @@ int append_bytes(uint8_t *buf, size_t len, const uint8_t *src, int src_len)
 
 	/* data fits into opal short atom */
 	if (src_len <= 15) {
-		if (len >= src_len + 1) {
+		if (len >= src_len + 1)
 			ret += append_short_atom_bytes_header(buf, len, src_len);
-			memcpy(buf + ret, src, src_len);
-			ret += src_len;
-		}
-	/* data fits into opal medium atom */
 	}
+
+	/* data fits into opal medium atom */
 	else if (src_len <= 2047) {
-		if (len >= src_len + 2) {
+		if (len >= src_len + 2)
 			ret += append_medium_atom_bytes_header(buf, len, src_len);
-			memcpy(buf + ret, src, src_len);
-			ret += src_len;
-		}
 	}
 
+	/* data fits into opal long atom */
+	else {
```

Comment:
long also has limit to 16777215B. suggesting to put condition checking it. and assert if somebody provides larger buffer otherwise

Reply:
Yeah I agree on that... will include the additional check